src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -125,12 +125,12 @@ public bool ChildContentRetrieved
         /// <summary>
         /// HTML attributes.
         /// </summary>
-        public TagHelperAttributeList HTMLAttributes { get; }
+        public TagHelperAttributeList HTMLAttributes => _htmlAttributes;
 
         /// <summary>
         /// <see cref="ITagHelper"/> bound attributes and HTML attributes.
         /// </summary>
-        public TagHelperAttributeList AllAttributes { get; }
+        public TagHelperAttributeList AllAttributes => _allAttributes;
```

Comment:
Reason why I don't put the `null` check in the getter is because of how this property is consumed when constructing a `TagHelperOutput`/`TagHelperContext` [here](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperRunner.cs#L44). 

The alternative would be to add public methods that indicate if attributes have been added `HtmlAttributesCreated`/`AllAttributesCreated`. Since `TagHelperExecutionContext`s will never be used by end-users I figured it wasn't worth the added API.